llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -570,6 +595,10 @@ bool AIEPostRASchedStrategy::isAvailableNode(SUnit &SU, SchedBoundary &Zone,
   if (isFixedSU(SU, !Zone.isTop()))
     return false;
 
+  // Instruction with delay slot should bever be scheduled in top-down.
+  if (Zone.isTop() && SU.getInstr()->hasDelaySlot())
+    return false;
+
```

Comment:
Nit: It would be nice to have a named predicate for this, like 'doesNotProgress' that is used both here and in the logic to switch to BottomUp